students/Nick_Lenssen/lesson03/mailroom.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+from sys import exit
+
+
+def create_report():
+    print("\n{:<22}{:<5}{:<20}{}{:<25}{}{:<15}".format(*('Donor Name','|','Total Given','|','Num Gifts','|','Average Gift')))
+    print ('-'*90)
+    for val, i in enumerate(main_list[0]):
+        print ("{:<20} {:>9} {:>9} {:>26}{:>11}{:>12}".format(*(i, '$', sum(main_list[val+1]), len(main_list[val+1]), '$',round(sum(main_list[val+1])/len(main_list[val+1]),1))))
+        
+def use_it(full_name):
+    while True:
+        amount = input("\nPlease enter in the amount you want to donate: ")
+        if amount == 'c':
+            main_list[0].pop()
+            main_list.pop()
+            print (main_list)
+            main()
```

Comment:
instead of adding donor and then asking amount and then potentially removing donor.. you can just ask amount first, ensure it's valid and then proceed with adding/updating donor.